CRM/Mailing/Service/ListUnsubscribe.php

```diff
@@ -54,6 +59,23 @@ public function alterMailParams(&$params, $context = NULL): void {
       return;
     }
 
+    $mailing_unsubscribe_mode = CRM_Core_DAO::singleValueQuery("SELECT m.unsubscribe_mode
+      FROM civicrm_mailing_event_queue mq
+      INNER JOIN civicrm_mailing m ON m.id = mq.mailing_id
+      WHERE mq.id = %1", [1 => [$m[2], 'Positive']]);
+    if ($mailing_unsubscribe_mode !== $mode) {
+      $mode = $mailing_unsubscribe_mode;
+    }
+    if ($methods === ['mailto']) {
```

Comment:
@JoeMurray yes it can be but if the only option selected is mailTo, other parts of the code base will have already added the unsubscribe email and as such the code in this class either doesn't need to do it in the case that the unsubscribe mode is unsubscribe but if it is opt out replace the value with the opt out email address and then do nothing else.